src/overlays/actors/ovl_En_Toto/z_en_toto.c

```diff
@@ -21,86 +60,703 @@ const ActorInit En_Toto_InitVars = {
+    func_80BA3DBC,
+};
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Toto_0x80BA36C0/EnToto_Draw.asm")
+void func_80BA36C0(EnToto* this, GlobalContext* globalCtx, s32 index) {
+    this->unk2B7 = 0;
+    this->actionFuncIndex = index;
+    D_80BA501C[this->actionFuncIndex](this, globalCtx);
+}
+
+void EnToto_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnToto* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    if (globalCtx->sceneNum == 0x15 && (gSaveContext.time >= 0x4000 && gSaveContext.time < 0xE555)) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 30.0f);
+    this->actor.bgCheckFlags |= 0x400;
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_0600A978, globalCtx->sceneNum == 0x12 ? &D_06003AA8 : &D_0600C880,
+                     this->limbDrawTbl, this->transitionDrawTbl, 18);
+    func_80BA36C0(this, globalCtx, 0);
+    this->actor.shape.rot.x = 0;
+}
+
+void EnToto_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnToto* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80BA383C(EnToto* this, GlobalContext* globalCtx) {
+    if (SkelAnime_FrameUpdateMatrix(&this->skelAnime) && this->actionFuncIndex == 1 &&
+        this->skelAnime.animCurrentSeg != &D_06000C80) {
+        if (globalCtx->msgCtx.unk11F04 != 0x2A98 && globalCtx->msgCtx.unk11F04 != 0x2A99) {
+            if (this->unk2B4 & 1 || Rand_ZeroOne() > 0.5f) {
+                this->unk2B4 = (this->unk2B4 + 1) & 3;
+            }
+        }
+        SkelAnime_ChangeAnimDefaultStop(&this->skelAnime, D_80BA5078[this->unk2B4]);
+    }
+    func_800BBB74(&this->unk260, 0x14, 0x50, 3);
+}
+
+void func_80BA3930(EnToto* this, GlobalContext* globalCtx) {
+    AnimationHeader* animationHeader;
+
+    animationHeader = &D_0600C880;
+    if (globalCtx->sceneNum == 0x12) {
+        animationHeader = &D_06003AA8;
+    }
+    SkelAnime_ChangeAnimTransitionRepeat(&this->skelAnime, animationHeader, -4.0f);
+}
+
+s32 func_80BA397C(EnToto* this, s16 arg1) {
+    s16 diff;
+
+    diff = this->actor.yawTowardsPlayer - this->actor.home.rot.y;
+    if (ABS_ALT(diff) < arg1) {
+        return 1;
+    }
+    return 0;
+}
+
+void func_80BA39C8(EnToto* this, GlobalContext* globalCtx) {
+    ActorPlayer* player = PLAYER;
+
+    func_80BA383C(this, globalCtx);
+    if (func_800B84D0(&this->actor, globalCtx) != 0) {
+        func_80BA36C0(this, globalCtx, 1);
+        if (globalCtx->sceneNum != 0x12) {
+            Actor_SetSwitchFlag(globalCtx, this->actor.params & 0x7F);
+        } else if (player->unk14B == 3) {
+            Actor_SetSwitchFlag(globalCtx, this->actor.home.rot.x);
+        }
+        this->unk2B6 = 0;
+        return;
+    }
+    if ((globalCtx->sceneNum == 0x15 && !(gSaveContext.time >= 0x4000 && gSaveContext.time < 0xED02)) ||
```

Comment:
```suggestion
    if ((globalCtx->sceneNum == 0x15 && (gSaveContext.time < 0x4000 || gSaveContext.time >= 0xED02)) ||
```
Would this work?

Reply:
Personally I prefer the not between two numbers rather then is less then a number or greater then a number. I find it easier to think about. Though if anyone has a strong opinion either way I'm willing to change it.

Reply:
No strong opinions from me. IMO do it however is natural to you.